src/app/archive/archive_lib/diff.ml

```diff
@@ -1,21 +1,121 @@
 open Coda_transition
 open Signature_lib
+open Core_kernel
 open Coda_base
+module Breadcrumb = Transition_frontier.Breadcrumb
 
+(* TODO: We should be able to fully deserialize and serialize via json *)
 module Transition_frontier = struct
-  type t =
+  open Transition_frontier.Diff
+
+  module Stable = struct
+    module V1 = struct
+      module T = struct
+        type t =
+          | Breadcrumb_added of
+              { block:
+                  ( External_transition.Stable.V1.t
+                  , State_hash.Stable.V1.t )
+                  With_hash.Stable.V1.t
+              ; sender_receipt_chains_from_parent_ledger:
+                  ( Public_key.Compressed.Stable.V1.t
+                  * Receipt.Chain_hash.Stable.V1.t )
+                  list }
+          | Root_transitioned of Root_transition.Lite.Stable.V1.t
+          | Bootstrap of {lost_blocks: State_hash.Stable.V1.t list}
+        [@@deriving bin_io, version]
```

Comment:
I think we want to use the new `[%%versioned` ppx instead of deriving `version`, right @psteckler ?